### PR TITLE
Add localized errors to version parsing.

### PR DESCRIPTION
**Pull request by BillyONeal**
Extracted from https://github.com/microsoft/vcpkg-tool/pull/404

### CODE REVIEW

File: include/vcpkg-test/util.h [hunk 1]
```diff
@@ -56,6 +57,14 @@ namespace Catch
     };
 }
 
+namespace vcpkg
+{
+    inline std::ostream& operator<<(std::ostream& os, const vcpkg::LocalizedString& value)
```
Comment (strega-nil) [inline include/vcpkg-test/util.h:62]:
```suggestion
    inline std::ostream& operator<<(std::ostream& os, const LocalizedString& value)
```

File: src/vcpkg/versions.cpp [hunk 2]
```diff
@@ -1,8 +1,34 @@
...
 #include <vcpkg/versions.h>
 
+namespace
+{
+    using namespace vcpkg;
+
+    DECLARE_AND_REGISTER_MESSAGE(VersionInvalidRelaxed,
+                                 (msg::version),
+                                 "",
+                                 "Error: String `{version}` is not a valid Relaxed version string (semver with "
```
Comment (strega-nil) [inline src/vcpkg/versions.cpp:15]:
```suggestion
                                 "string `{version}` is not a valid relaxed version string (semver with "
```

you should use `msg::msgErrorMessage` (or w/e we call it)

File: src/vcpkg/versions.cpp [hunk 3]
```diff
@@ -1,8 +1,34 @@
...
+                                 "arbitrary numeric element count)");
+
+    DECLARE_AND_REGISTER_MESSAGE(
+        VersionInvalidSemver,
+        (msg::version),
+        "",
+        "Error: String `{version}` is not a valid Semantic Version string, consult https://semver.org");
```
Comment (strega-nil) [inline src/vcpkg/versions.cpp:22]:
```suggestion
        "string `{version}` is not a valid semantic version string, consult https://semver.org");
```

File: src/vcpkg/versions.cpp [hunk 4]
```diff
@@ -1,8 +1,34 @@
...
+
+    DECLARE_AND_REGISTER_MESSAGE(VersionInvalidDate,
+                                 (msg::version),
+                                 "",
+                                 "Error: String `{version}` is not a valid date version."
```
Comment (strega-nil) [inline src/vcpkg/versions.cpp:27]:
```suggestion
                                 "string '{version}' is not a valid date version. "
```

File: src/vcpkg/versions.cpp [hunk 5]
```diff
@@ -251,8 +270,7 @@ namespace vcpkg
             }
         }
 
-        return Strings::format("Error: String `%s` is not a valid Semantic Version string, consult https://semver.org",
-                               str);
+        return msg::format(msgVersionInvalidSemver, msg::version = str);
```
Comment (strega-nil) [inline src/vcpkg/versions.cpp:273]:
```suggestion
        return msg::format(msg::msgErrorMessage).append(msgVersionInvalidSemver, msg::version = str);
```

**Review by strega-nil — COMMENTED** [review]
mostly minor nits. The space after the period is important tho!

File: src/vcpkg/versions.cpp (hunk 4 above)
Comment (BillyONeal) [reply src/vcpkg/versions.cpp:27]: >The space after the period is important tho!

Why?

**Review by strega-nil-ms — APPROVED** [review]

**Pull request merged**